modules/pureconfig/shared/src/main/scala-3.0+/eu/timepit/refined/pureconfig/TypeDescribeVersionSpecific.scala

```diff
@@ -0,0 +1,19 @@
+package eu.timepit.refined.pureconfig
+
+import scala.quoted.{Expr, Quotes, Type, quotes}
+
+final class DescribeType[T](val repr: String)
+
+object DescribeType:
+
+  private def macroImpl[T: Type](using Quotes): Expr[DescribeType[T]] = {
+    import quotes.reflect.*
+    '{ new DescribeType[T](${ Expr(TypeRepr.of[T].show) }) }
+  }
+
+  inline given describeType[T]: DescribeType[T] = ${ macroImpl[T] }
+
+private[pureconfig] trait TypeDescribeVersionSpecific {
+
+  protected def getDescription[A](d: DescribeType[A]): String = d.repr
+}
```

Comment:
This looks similar to [`TypeName` in ciris](https://github.com/vlovgr/ciris/blob/v3.1.0/modules/refined/shared/src/main/scala-3/ciris/refined/internal/TypeNameRuntimePlatform.scala) and the (unmaintained?) https://github.com/tpolecat/typename project.

Is there a difference between `TypeRepr.of[T].show` as used here and `Type.show[A]`? According to the [docs](https://github.com/lampepfl/dotty/blob/3.2.2/library/src/scala/quoted/Type.scala#LL17C5-L17C24)  `Type.show[A]` is the same as `TypeTree.of[T].show`.

Reply:
@fthomas The difference is between `Type[T]` and `TypeRepr[T]` - they're different thing, but I'm not sure we care about differences here.
I've found [this article](https://softwaremill.com/scala-3-macros-tips-and-tricks/) extremely helpful, I've used it as a starting point when implementing, there's a note about both `Type` and `TypeRepr`

As for `show`ing - it's all about using `Printer`s - but `Type.show[A]` signature is fixed to the default one, while `TypeRepr[T].show` accepts a given `Printer` (if not in scope, resolves to `Printer.TypeReprCode`). 

So, finally -

 - `Type.show[A]` is `Printer.TreeCode.show(summon[TypeTree[T]])`
 - `TypeRepr.of[T].show` is `Printer.TypeReprCode.show(summon[TypeRepr[T]])`

Other available printers can be found here in [dotty sources](https://github.com/lampepfl/dotty/blob/528d93189480570532ee20b27814dcc16a014775/library/src/scala/quoted/Quotes.scala#L5057-L5086) - let me know if it makes sense to try another one for prettier results

Reply:
Many thanks for the explanation and links, @ivan-klass! There is no need to try another printer here. I just hope that in the future this kind of macro does not need to be reinvented in every library. :crossed_fingers: